### ISSUE TITLE
Crash while viewing several statistics

### ISSUE THREAD

**Issue by nolyn** [IENT/YUView#41]
When I open several statistics YUView crashes suddenly, when changing between the statistics. It occurs after a random time, not at the first change. I tried to trace it in Debugging mode, but couldn't find the location. I used these statistics:

/data/sauer/simulations/PIIdeas/DirectionalInpainting/UHasseltSoccer/FTBE20/WUHSwindowsize9_DirInpIgnoreFG_All_bdkw/Soccer_Arc_1920x1080_25_2V_baseCfg_2view+depth_allintra_FTBE20_QP30+39+30+39stats_2.csv
/data/sauer/simulations/PIIdeas/DirectionalInpainting/UHasseltSoccer/FTBE20/WUHSwindowsize9_DirInpIgnoreFG_All_bdkw/Soccer_Arc_1920x1080_25_2V_baseCfg_2view+depth_allintra_FTBE20_QP30+39+30+39stats_3.csv

the backtrace:

Thread 18 (Thread 0x7fff9bfff700 (LWP 25708)):
#0  0x00007ffff4b4a48d in poll () from /usr/lib/libc.so.6

No symbol table info available.
#1  0x00007ffff293a066 in ?? () from /usr/lib/libglib-2.0.so.0

No symbol table info available.
#2  0x00007ffff293a17c in g_main_context_iteration () from /usr/lib/libglib-2.0.so.0

No symbol table info available.
#3  0x00007fffb820b4bd in ?? () from /usr/lib/gio/modules/libdconfsettings.so

No symbol table info available.
#4  0x00007ffff2960345 in ?? () from /usr/lib/libglib-2.0.so.0

No symbol table info available.
#5  0x00007ffff58e1454 in start_thread () from /usr/lib/libpthread.so.0

No symbol table info available.
#6  0x00007ffff4b537df in clone () from /usr/lib/libc.so.6

No symbol table info available.

Thread 8 (Thread 0x7fffcec8e700 (LWP 25669)):
#0  0x00007ffff4b4a48d in poll () from /usr/lib/libc.so.6

No symbol table info available.
#1  0x00007ffff293a066 in ?? () from /usr/lib/libglib-2.0.so.0

No symbol table info available.
#2  0x00007ffff293a17c in g_main_context_iteration () from /usr/lib/libglib-2.0.so.0

No symbol table info available.
#3  0x00007ffff604659b in QEventDispatcherGlib::processEvents(QFlagsQEventLoop::ProcessEventsFlag) () from /usr/lib/libQt5Core.so.5

No symbol table info available.
#4  0x00007ffff5ff00da in QEventLoop::exec(QFlagsQEventLoop::ProcessEventsFlag) () from /usr/lib/libQt5Core.so.5

No symbol table info available.
#5  0x00007ffff5e130f3 in QThread::exec() () from /usr/lib/libQt5Core.so.5

No symbol table info available.
#6  0x00007ffff5e17d78 in ?? () from /usr/lib/libQt5Core.so.5

No symbol table info available.
#7  0x00007ffff58e1454 in start_thread () from /usr/lib/libpthread.so.0

No symbol table info available.
#8  0x00007ffff4b537df in clone () from /usr/lib/libc.so.6

No symbol table info available.

Thread 6 (Thread 0x7fffdcc55700 (LWP 25660)):
#0  0x00007ffff4b4a48d in poll () from /usr/lib/libc.so.6

No symbol table info available.
#1  0x00007ffff293a066 in ?? () from /usr/lib/libglib-2.0.so.0

No symbol table info available.
#2  0x00007ffff293a17c in g_main_context_iteration () from /usr/lib/libglib-2.0.so.0

No symbol table info available.
#3  0x00007ffff604657f in QEventDispatcherGlib::processEvents(QFlagsQEventLoop::ProcessEventsFlag) () from /usr/lib/libQt5Core.so.5

No symbol table info available.
#4  0x00007ffff5ff00da in QEventLoop::exec(QFlagsQEventLoop::ProcessEventsFlag) () from /usr/lib/libQt5Core.so.5

No symbol table info available.
#5  0x00007ffff5e130f3 in QThread::exec() () from /usr/lib/libQt5Core.so.5

No symbol table info available.
#6  0x00007ffff5e17d78 in ?? () from /usr/lib/libQt5Core.so.5

No symbol table info available.
#7  0x00007ffff58e1454 in start_thread () from /usr/lib/libpthread.so.0

No symbol table info available.
#8  0x00007ffff4b537df in clone () from /usr/lib/libc.so.6

No symbol table info available.

Thread 5 (Thread 0x7fffde138700 (LWP 25658)):
#0  0x00007ffff4b4a48d in poll () from /usr/lib/libc.so.6

No symbol table info available.
#1  0x00007ffff293a066 in ?? () from /usr/lib/libglib-2.0.so.0

No symbol table info available.
#2  0x00007ffff293a17c in g_main_context_iteration () from /usr/lib/libglib-2.0.so.0

No symbol table info available.
#3  0x00007ffff604659b in QEventDispatcherGlib::processEvents(QFlagsQEventLoop::ProcessEventsFlag) () from /usr/lib/libQt5Core.so.5

No symbol table info available.
#4  0x00007ffff5ff00da in QEventLoop::exec(QFlagsQEventLoop::ProcessEventsFlag) () from /usr/lib/libQt5Core.so.5

No symbol table info available.
#5  0x00007ffff5e130f3 in QThread::exec() () from /usr/lib/libQt5Core.so.5

No symbol table info available.
#6  0x00007fffea7c4e75 in ?? () from /usr/lib/libQt5DBus.so.5

No symbol table info available.
#7  0x00007ffff5e17d78 in ?? () from /usr/lib/libQt5Core.so.5

No symbol table info available.
#8  0x00007ffff58e1454 in start_thread () from /usr/lib/libpthread.so.0

No symbol table info available.
#9  0x00007ffff4b537df in clone () from /usr/lib/libc.so.6

No symbol table info available.

Thread 4 (Thread 0x7fffdfbbf700 (LWP 25657)):
#0  0x00007ffff4b4a48d in poll () from /usr/lib/libc.so.6

No symbol table info available.
#1  0x00007ffff293a066 in ?? () from /usr/lib/libglib-2.0.so.0

No symbol table info available.
#2  0x00007ffff293a3f2 in g_main_loop_run () from /usr/lib/libglib-2.0.so.0

No symbol table info available.
#3  0x00007fffe28927c6 in ?? () from /usr/lib/libgio-2.0.so.0

No symbol table info available.
#4  0x00007ffff2960345 in ?? () from /usr/lib/libglib-2.0.so.0

No symbol table info available.
#5  0x00007ffff58e1454 in start_thread () from /usr/lib/libpthread.so.0

No symbol table info available.
#6  0x00007ffff4b537df in clone () from /usr/lib/libc.so.6

No symbol table info available.

Thread 3 (Thread 0x7fffe03c0700 (LWP 25656)):
#0  0x00007ffff4b4a48d in poll () from /usr/lib/libc.so.6

No symbol table info available.
#1  0x00007ffff293a066 in ?? () from /usr/lib/libglib-2.0.so.0

No symbol table info available.
#2  0x00007ffff293a17c in g_main_context_iteration () from /usr/lib/libglib-2.0.so.0

No symbol table info available.
#3  0x00007ffff293a1c1 in ?? () from /usr/lib/libglib-2.0.so.0

No symbol table info available.
#4  0x00007ffff2960345 in ?? () from /usr/lib/libglib-2.0.so.0

No symbol table info available.
#5  0x00007ffff58e1454 in start_thread () from /usr/lib/libpthread.so.0

No symbol table info available.
#6  0x00007ffff4b537df in clone () from /usr/lib/libc.so.6

No symbol table info available.

Thread 2 (Thread 0x7fffea153700 (LWP 25648)):
#0  0x00007ffff4b4a48d in poll () from /usr/lib/libc.so.6

No symbol table info available.
#1  0x00007ffff0f0b8e0 in ?? () from /usr/lib/libxcb.so.1

No symbol table info available.
#2  0x00007ffff0f0d679 in xcb_wait_for_event () from /usr/lib/libxcb.so.1

No symbol table info available.
#3  0x00007fffec982469 in ?? () from /usr/lib/libQt5XcbQpa.so.5

No symbol table info available.
#4  0x00007ffff5e17d78 in ?? () from /usr/lib/libQt5Core.so.5

No symbol table info available.
#5  0x00007ffff58e1454 in start_thread () from /usr/lib/libpthread.so.0

No symbol table info available.
#6  0x00007ffff4b537df in clone () from /usr/lib/libc.so.6

No symbol table info available.

Thread 1 (Thread 0x7ffff7f56cc0 (LWP 25630)):
#0  0x00007fffeca0296a in QFontEngineFT::loadGlyphSet(QTransform const&) () from /usr/lib/libQt5XcbQpa.so.5

No symbol table info available.
#1  0x00007fffeca029e9 in QFontEngineFT::loadGlyphFor(unsigned int, QFixed, QFontEngine::GlyphFormat, QTransform const&, bool) () from /usr/lib/libQt5XcbQpa.so.5

No symbol table info available.
#2  0x00007fffeca041f5 in QFontEngineFT::lockedAlphaMapForGlyph(unsigned int, QFixed, QFontEngine::GlyphFormat, QTransform const&, QPoint*) () from /usr/lib/libQt5XcbQpa.so.5

No symbol table info available.
#3  0x00007ffff6f36d20 in QRasterPaintEngine::drawCachedGlyphs(int, unsigned int const_, QFixedPoint const_, QFontEngine*) () from /usr/lib/libQt5Gui.so.5

No symbol table info available.
#4  0x00007ffff6f3ada7 in QRasterPaintEngine::drawTextItem(QPointF const&, QTextItem const&) () from /usr/lib/libQt5Gui.so.5

No symbol table info available.
#5  0x00007ffff6f573c6 in ?? () from /usr/lib/libQt5Gui.so.5

No symbol table info available.
#6  0x00007ffff6dd647e in QTextLine::draw(QPainter_, QPointF const&, QTextLayout::FormatRange const_) const () from /usr/lib/libQt5Gui.so.5

No symbol table info available.
#7  0x00007ffff6dd727d in QTextLayout::draw(QPainter*, QPointF const&, QVectorQTextLayout::FormatRange const&, QRectF const&) const () from /usr/lib/libQt5Gui.so.5

No symbol table info available.
#8  0x00007ffff764b27b in QWidgetLineControl::draw(QPainter*, QPoint const&, QRect const&, int) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#9  0x00007ffff75ba748 in QLineEdit::paintEvent(QPaintEvent*) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#10 0x00007ffff74c3f18 in QWidget::event(QEvent*) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#11 0x00007ffff75bf399 in QLineEdit::event(QEvent*) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#12 0x00007ffff747ce3c in QApplicationPrivate::notify_helper(QObject_, QEvent_) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#13 0x00007ffff74845b1 in QApplication::notify(QObject_, QEvent_) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#14 0x00007ffff5ff1c80 in QCoreApplication::notifyInternal2(QObject_, QEvent_) () from /usr/lib/libQt5Core.so.5

No symbol table info available.
#15 0x00007ffff74bccea in QWidgetPrivate::sendPaintEvent(QRegion const&) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#16 0x00007ffff74bd339 in QWidgetPrivate::drawWidget(QPaintDevice_, QRegion const&, QPoint const&, int, QPainter_, QWidgetBackingStore*) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#17 0x00007ffff74be02c in QWidgetPrivate::paintSiblingsRecursive(QPaintDevice_, QList<QObject_> const&, int, QRegion const&, QPoint const&, int, QPainter_, QWidgetBackingStore_) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#18 0x00007ffff74bcea4 in QWidgetPrivate::drawWidget(QPaintDevice_, QRegion const&, QPoint const&, int, QPainter_, QWidgetBackingStore*) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#19 0x00007ffff74be02c in QWidgetPrivate::paintSiblingsRecursive(QPaintDevice_, QList<QObject_> const&, int, QRegion const&, QPoint const&, int, QPainter_, QWidgetBackingStore_) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#20 0x00007ffff74bdf19 in QWidgetPrivate::paintSiblingsRecursive(QPaintDevice_, QList<QObject_> const&, int, QRegion const&, QPoint const&, int, QPainter_, QWidgetBackingStore_) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#21 0x00007ffff74bdf19 in QWidgetPrivate::paintSiblingsRecursive(QPaintDevice_, QList<QObject_> const&, int, QRegion const&, QPoint const&, int, QPainter_, QWidgetBackingStore_) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#22 0x00007ffff74bdf19 in QWidgetPrivate::paintSiblingsRecursive(QPaintDevice_, QList<QObject_> const&, int, QRegion const&, QPoint const&, int, QPainter_, QWidgetBackingStore_) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#23 0x00007ffff74bdf19 in QWidgetPrivate::paintSiblingsRecursive(QPaintDevice_, QList<QObject_> const&, int, QRegion const&, QPoint const&, int, QPainter_, QWidgetBackingStore_) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#24 0x00007ffff74bdf19 in QWidgetPrivate::paintSiblingsRecursive(QPaintDevice_, QList<QObject_> const&, int, QRegion const&, QPoint const&, int, QPainter_, QWidgetBackingStore_) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#25 0x00007ffff74bcea4 in QWidgetPrivate::drawWidget(QPaintDevice_, QRegion const&, QPoint const&, int, QPainter_, QWidgetBackingStore*) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#26 0x00007ffff74be02c in QWidgetPrivate::paintSiblingsRecursive(QPaintDevice_, QList<QObject_> const&, int, QRegion const&, QPoint const&, int, QPainter_, QWidgetBackingStore_) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#27 0x00007ffff74bcea4 in QWidgetPrivate::drawWidget(QPaintDevice_, QRegion const&, QPoint const&, int, QPainter_, QWidgetBackingStore*) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#28 0x00007ffff74be02c in QWidgetPrivate::paintSiblingsRecursive(QPaintDevice_, QList<QObject_> const&, int, QRegion const&, QPoint const&, int, QPainter_, QWidgetBackingStore_) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#29 0x00007ffff74bcea4 in QWidgetPrivate::drawWidget(QPaintDevice_, QRegion const&, QPoint const&, int, QPainter_, QWidgetBackingStore*) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#30 0x00007ffff74be02c in QWidgetPrivate::paintSiblingsRecursive(QPaintDevice_, QList<QObject_> const&, int, QRegion const&, QPoint const&, int, QPainter_, QWidgetBackingStore_) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#31 0x00007ffff74bcea4 in QWidgetPrivate::drawWidget(QPaintDevice_, QRegion const&, QPoint const&, int, QPainter_, QWidgetBackingStore*) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#32 0x00007ffff74be02c in QWidgetPrivate::paintSiblingsRecursive(QPaintDevice_, QList<QObject_> const&, int, QRegion const&, QPoint const&, int, QPainter_, QWidgetBackingStore_) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#33 0x00007ffff74bdf19 in QWidgetPrivate::paintSiblingsRecursive(QPaintDevice_, QList<QObject_> const&, int, QRegion const&, QPoint const&, int, QPainter_, QWidgetBackingStore_) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#34 0x00007ffff74bdf19 in QWidgetPrivate::paintSiblingsRecursive(QPaintDevice_, QList<QObject_> const&, int, QRegion const&, QPoint const&, int, QPainter_, QWidgetBackingStore_) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#35 0x00007ffff74bdf19 in QWidgetPrivate::paintSiblingsRecursive(QPaintDevice_, QList<QObject_> const&, int, QRegion const&, QPoint const&, int, QPainter_, QWidgetBackingStore_) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#36 0x00007ffff74bcea4 in QWidgetPrivate::drawWidget(QPaintDevice_, QRegion const&, QPoint const&, int, QPainter_, QWidgetBackingStore*) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#37 0x00007ffff748d0fa in ?? () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#38 0x00007ffff748d2b7 in ?? () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#39 0x00007ffff74ac11f in QWidgetPrivate::syncBackingStore() () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#40 0x00007ffff74c3fe8 in QWidget::event(QEvent*) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#41 0x00007ffff75c250b in QMainWindow::event(QEvent*) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#42 0x00007ffff747ce3c in QApplicationPrivate::notify_helper(QObject_, QEvent_) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#43 0x00007ffff74845b1 in QApplication::notify(QObject_, QEvent_) () from /usr/lib/libQt5Widgets.so.5

No symbol table info available.
#44 0x00007ffff5ff1c80 in QCoreApplication::notifyInternal2(QObject_, QEvent_) () from /usr/lib/libQt5Core.so.5

No symbol table info available.
#45 0x00007ffff5ff43fd in QCoreApplicationPrivate::sendPostedEvents(QObject_, int, QThreadData_) () from /usr/lib/libQt5Core.so.5

No symbol table info available.
#46 0x00007ffff6046173 in ?? () from /usr/lib/libQt5Core.so.5

No symbol table info available.
#47 0x00007ffff2939e67 in g_main_context_dispatch () from /usr/lib/libglib-2.0.so.0

No symbol table info available.
#48 0x00007ffff293a0d0 in ?? () from /usr/lib/libglib-2.0.so.0

No symbol table info available.
#49 0x00007ffff293a17c in g_main_context_iteration () from /usr/lib/libglib-2.0.so.0

No symbol table info available.
#50 0x00007ffff604657f in QEventDispatcherGlib::processEvents(QFlagsQEventLoop::ProcessEventsFlag) () from /usr/lib/libQt5Core.so.5

No symbol table info available.
#51 0x00007ffff5ff00da in QEventLoop::exec(QFlagsQEventLoop::ProcessEventsFlag) () from /usr/lib/libQt5Core.so.5

No symbol table info available.
#52 0x00007ffff5ff85cc in QCoreApplication::exec() () from /usr/lib/libQt5Core.so.5

No symbol table info available.
#53 0x00000000004abda4 in main (argc=1, argv=0x7fffffffe408) at ../YUView_development/source/yuviewapp.cpp:84

```
    a = {<QApplication> = {<No data fields>}, static staticMetaObject = {d = {superdata = 0x7ffff7b531e0 <QApplication::staticMetaObject>, stringdata = 0x4f4ec0 <qt_meta_stringdata_YUViewApp>, data = 0x4f4f00 <qt_meta_data_YUViewApp>, static_metacall = 0x4b2b1c <YUViewApp::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, extradata = 0x0}}, w = 0x877a50}
```

basically the debugger shows the line   return a.exec();
everything else is already inside the QT libraries

**Comment by ChristianFeldmann** [comment]
I can't get this to crash. Maybe it has something to do with the Qt Version?

**Comment by nolyn** [comment]
Hab das Problem unter Linux:
Laptop: QT 5.7.0-2
Institutsrechner: QT 5.5.1

Es scheint jedoch auch vom Statistics File abzuhängen. Mit einem anderen hatte ich gar keine Probleme. Noch nicht geschafft zu traces woran es liegt

**Comment by JensAc** [comment]
I think, the problem is related how qt interacts with fonts on linux. I changed the interface font to something non-default and I did not get the crash in the first place anymore. no guaranties. Please try and comment.

**Comment by JensAc** [comment]
Nope....got another crash today. The crash happens in QFontEngineFT::loadGlyphSet(QTransform const&)

**Comment by JensAc** [comment]
I also get a bunch of 
QWidget::setLayout: Attempting to set QLayout "verticalLayout" on QWidget "playlistItemStatisticsFile", which already has a layout
warnings before the fault happens. Maybe this is related?!

**Comment by JensAc** [comment]
See
[(http://stackoverflow.com/questions/40490414/cannot-trace-cause-of-crash-in-qt-program)]

**Comment by KubaO** [comment]
The setLayout warnings are real, but they shouldn't cause crashes although they are brittle. I have all of them fixed in my tree, will get a pull request for it tomorrow. There is a possibility for undetected dangling pointers here and there, I've also fixed that.

**Comment by nolyn** [comment]
It still crashed for me, but I'm using the workaround with the environment
variable, then it works fine.

**Comment by ChristianFeldmann** [comment]
Cleaning up the issues. The issue is quite old. If this also occurs with the new version, please let us know!
